include/zephyr/arch/arc/v2/linker.ld

```diff
@@ -58,7 +60,12 @@ MEMORY {
 	ICCM  (rwx)  : ORIGIN = ICCM_START,  LENGTH = ICCM_SIZE
 #endif
 #ifdef SRAM_START
+#if defined(CONFIG_XIP)
 	SRAM  (rwx) : ORIGIN = SRAM_START,  LENGTH = SRAM_SIZE
+#else
+	SRAM_RX  (rx) : ORIGIN = SRAM_START,  LENGTH = (SRAM_SIZE / 2)
+	SRAM_RW  (rw) : ORIGIN = SRAM_START + (SRAM_SIZE / 2),  LENGTH = (SRAM_SIZE / 2)
```

Comment:
Hmm, I'm wondering if the `LENGTH` is actually used somehow? 

The issue is that the ratio between `SRAM_RX` and `SRAM_RW` may differ an most likely it won't be 1:1.
Is it possible to avoid defining `LENGTH`?

Reply:
If it's possible to not define `LENGTH` we can just do an assert of combined SRAM_RX and SRAM_RW sizes against SRAM size in the end.

Reply:
As far as I understand "LENGTH" is default keyword in description of memory blocks in scripts. And it should be defined for correct script working

Reply:
I believe the issue is not about "LENGTH", it is about the fixed half-half split. Since from the build process it should be known what the text and data sizes are, can't you avoid introducing new fixed-size linker segments?
Introduce some symbols that are correctly aligned (if not available already) would be enough, and use these to program the MPU.

Reply:
Thanks, current approach makes more sense (though needs some more work/though I believe).